File: frontend/src/app/service/userDTO/userDTO.ts

interface UserDTO {
    Id: number;
    Name: string;
    Email: string;
    Username: string;
    Password: string;
    Roles: RoleDTO[];
}
  
interface RoleDTO {
    Id: number;
    Name: string;
}

interface SquadDTO {
    Id: number;
    Name: string;
    Description: string;
}

interface UserLoginDTO {
    Username: string;
    Password: string;
}

File: backend/UserAPI/src/API/DTOs/UserDTO.cs
using Squad.API.DTOs;

namespace UserAuth.API.DTOs
{
    public class UserDTO
    {
        public int? Id { get; set; }
        public string? Name { get; set; }
        public string? Email { get; set; }
        public string? Username { get; set; }
        public string? Password { get; set; }
        public List<RoleDTO> Roles { get; set; } = new List<RoleDTO>();

        public List<SquadDTO> Squads { get; set; } = new List<SquadDTO>();
    }
}